pkg/cache/cluster.go

```diff
@@ -1240,16 +1296,65 @@ func (c *clusterCache) processEvent(event watch.EventType, un *unstructured.Unst
 		return
 	}
 
+	c.eventMetaCh <- eventMeta{event, un}
+}
+
+func (c *clusterCache) processEvents() {
+	log := c.log.WithValues("fn", "processItems")
+	log.V(1).Info("Start processing events")
+
 	c.lock.Lock()
-	defer c.lock.Unlock()
-	existingNode, exists := c.resources[key]
-	if event == watch.Deleted {
-		if exists {
-			c.onNodeRemoved(key)
+	ch := c.eventMetaCh
+	c.lock.Unlock()
+
+	eventMetas := make([]eventMeta, 0)
+	ticker := time.NewTicker(c.eventProcessingInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case em, ok := <-ch:
```

Comment:
Use a more verbose name like `eventMeta`. `em` is not a standard shortening.

Reply:
Declaration of `eventMeta` shadows declaration at `cluster.go` - https://github.com/argoproj/gitops-engine/pull/629/files#diff-9c9e197d543705f08c9b1bc2dc404a55506cfc2935a988e6007d248257aadb1aR80.

Renamed to `evMeta`.